### PR TITLE
looking-glass-client: new module

### PR DESCRIPTION
**Pull request by j-brn**
### Description

Added a module to generate the configuration file for looking-glass-client

### Checklist

- [x] Change is backwards compatible. 

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/looking-glass-client.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            rawMouse = true;
+            escapeKey = 62;
+          };
+        }
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
```
Comment (polykernel) [inline modules/programs/looking-glass-client.nix:56]:
```suggestion
    assertions =
      [ (hm.assertions.assertPlatform "programs.looking-glass-client" pkgs platforms.linux) ];

    home.packages = [ cfg.package ];
```

**Review by polykernel — APPROVED** [review]

File: modules/programs/looking-glass-client.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    package = mkOption {
+      default = pkgs.looking-glass-client;
+      defaultText = literalExpression "pkgs.looking-glass-client";
+      type = types.package;
+      example = literalExpression "pkgs.another-looking-glass-client-package";
+      description = ''
+        Looking-Glass-Client package to be used.
+      '';
+    };
```
Comment (berbiche) [inline modules/programs/looking-glass-client.nix:22]: Small nit.
```suggestion
    package = mkPackageOption pkgs "looking-glass-client" { };
```

**Review by berbiche — APPROVED** [review]
Thanks for the contribution!

**Comment by j-brn** [comment]
> Hi @j-brn, would you mind running the formatter script? `./format` from the root of your fork

sure. -> 8a539df77e7edef3ba1d3ea925874df22d681441

File: modules/programs/looking-glass-client.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  options.programs.looking-glass-client = {
+    enable = mkEnableOption "looking-glass-client";
+
+    package = mkPackageOption pkgs "looking-glass-client" { };
+
+    settings = mkOption {
+      type = settingsFormat.type;
+      default = { };
+      description = "looking-glass-client settings";
```
Comment (berbiche) [inline modules/programs/looking-glass-client.nix:19]:
```suggestion
      description = "looking-glass-client settings.";
```
Missed the final dot.

**Comment by rycee** [comment]
Thanks! Rebased to master in 42f81ac107c5a9a177888080107094dba57f134e.

**Pull request closed without merging**